packages/flutter/lib/src/widgets/form.dart

```diff
@@ -191,6 +191,18 @@ class FormState extends State<Form> {
       hasError = !field.validate() || hasError;
     return !hasError;
   }
+
+  /// Validates every [FormField] that is a descendant of this [Form], and
+  /// returns the list of invalid fields, or an empty list if there are none.
+  ///
+  /// The form will rebuild to report the results.
+  List<FormFieldState<dynamic>> invalidFields() {
+    _hasInteractedByUser = true;
+    _forceRebuild();
```

Comment:
I worry a little bit about setting _hasInteractedByUser to true when the user actually has not interacted with it.  I guess if you don't include these two lines then it's possible to get out of date results from field.validate()?

Reply:
I guess this comes down to making the choice if this function runs the validators on all the formfields within this form, or just reports the state as it was when this function is called. Since `FormFieldState.validate` rebuilds the field, it seems correct to force a rebuild of the entire form. 

I thought of this function as doing the same what `validate` does, and has the added bonus of knowing which fields are incorrect afterwards. The naming of this function may need some adaptations in that case